java/idl2jni/codegen/im_jni.cpp

```diff
@@ -1791,6 +1795,30 @@ bool idl_mapping_jni::gen_union(UTL_ScopedName *name,
       "  " + disc_ty + " jdisc = source._d ();\n";
   }
 
+  if (!hasDefault && default_value.computed_ != 0) {
```

Comment:
Why is hasDefault negated?

Reply:
This is from the IDL-to-Java spec (IDL-to-C++ is very similar):
> Two default modifier methods, both named __default(), are generated if there is no explicit default case label, and the set of case labels does not completely cover the possible values of the discriminant.

Reply:
So I'm assuming that `default_value.computed_ != 0` is a way of detecting if the case labels don't completely cover the values of the discriminant?

Reply:
https://github.com/DOCGroup/ACE_TAO/blob/7c298ab69026c64c027d63530f7ca275f0bc9c73/TAO/TAO_IDL/include/ast_union.h#L122